lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -0,0 +1,354 @@
+          post[:card][:present] = options[:card][:present]
+          post[:card][:securityCode] = {} unless post[:card][:securityCode]
+          post[:card][:securityCode][:indicator] = options[:card][:indicator] if options[:card][:indicator]
+          post[:card][:securityCode][:value] = options[:card][:value] if options[:card][:value]
+          post[:card][:securityCode] = {}
+        end
+        if options[:security_code].present?
+          post[:card][:securityCode] = {}
+          post[:card][:securityCode][:indicator] = options[:security_code][:indicator]
+          post[:card][:securityCode][:value] = options[:security_code][:value]
+        end
+      end
+
+      def add_customer(post, options)
+        if address = options[:customer]
+          post[:customer] = {}
+          post[:customer][:addressLine1] = address[:address_line1] if address[:address_line1]
+          post[:customer][:firstName] = address[:first_name]
+          post[:customer][:lastName] = address[:last_name]
+          post[:customer][:postalCode] = address[:postal_code]
+        end
+      end
+
+      def add_purchase_card(post, options)
+        post[:transaction] = {}
+        post[:transaction][:invoice] = options[:invoice]
+        post[:transaction][:notes] = options[:notes] if options[:notes].present?
+        if options[:purchase_card].present?
+          post[:transaction][:purchaseCard] = {}
+          post[:transaction][:purchaseCard][:customerReference] = options[:purchase_card][:customer_reference]
+          post[:transaction][:purchaseCard][:destinationPostalCode] = options[:purchase_card][:destination_postal_code]
+          post[:transaction][:purchaseCard][:productDescriptors] = options[:purchase_card][:product_descriptors]
+        end
+      end
+
+      def add_override_bus_date(post, options)
+        post[:overrideBusDate] = options[:override_bus_date] if options[:override_bus_date].present?
+      end
+
+      def add_amex(post, options)
+        if options[:amex].present?
+          post[:amex] = {}
+          post[:amex][:propertyCode] = options[:amex][:property_code]
+        end
+      end
+
+      def add_auto(post, options)
+        if options[:auto].present?
+          post[:auto] = {}
+          post[:auto][:estimatedDays] = options[:auto][:estimated_days]
+        end
+      end
+
+      def add_card_on_file(post, options)
+        if options[:stored_credential].present?
+          post[:cardOnFile] = {}
+          post[:cardOnFile][:indicator] = options[:stored_credential][:indicator] if options[:stored_credential][:indicator]
+          if usage_indicator = options[:stored_credential][:usage_indicator] == '01' ? 'Initial' : 'Subsequent'
+            post[:cardOnFile][:usageIndicator] = usage_indicator
+          end
+          if scheduled_indicator = options[:stored_credential][:scheduled_indicator] == '01' ? 'Scheduled' : 'Unscheduled'
+            post[:cardOnFile][:scheduledIndicator] = scheduled_indicator
+          end
```

Comment:
It looks like your checks are backwards here. You'll want to check if the `options[:stored_credential][:schedule_indicator] == 'scheduled' ...` and then assign the `01` (which looks to be the Shift4 value) to the variable that is added to the `post` body.